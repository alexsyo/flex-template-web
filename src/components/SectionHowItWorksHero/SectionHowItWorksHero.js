import React from 'react';
import { string } from 'prop-types';
import { FormattedMessage } from 'react-intl';
import classNames from 'classnames';
import { NamedLink } from '..';

import css from './SectionHowItWorksHero.css';

const SectionHowItWorksHero = props => {
  const { rootClassName, className } = props;

  const classes = classNames(rootClassName || css.root, className);

  return (
    <div className={classes}>
      <div className={css.heroContent}>
        <h1 className={css.heroMainTitle}>
          <FormattedMessage id="SectionHowItWorksHero.title" />
        </h1>
        <h2 className={css.heroSubTitle}>
          <FormattedMessage id="SectionHowItWorksHero.subTitle" />
        </h2>
        <NamedLink
          name="SearchPage"
          to={{
            search:
              'address=Netherlands&bounds=53.665238%2C7.230902%2C50.750667%2C3.1862592',
          }}
          className={css.heroButton}
        >
          <FormattedMessage id="SectionHowItWorksHero.browseButton" />
        </NamedLink>
      </div>
    </div>
  );
};

SectionHowItWorksHero.defaultProps = { rootClassName: null, className: null };

SectionHowItWorksHero.propTypes = {
  rootClassName: string,
  className: string,
};

export default SectionHowItWorksHero;
