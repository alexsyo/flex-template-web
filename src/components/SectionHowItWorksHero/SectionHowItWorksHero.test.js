import React from 'react';
import { renderDeep } from '../../util/test-helpers';
import SectionHowItWorksHero from './SectionHowItWorksHero';

const noop = () => null;

describe('SectionHowItWorksHero', () => {
  it('matches snapshot', () => {
    window.google = { maps: {} };
    const heroProps = {
      history: { push: noop },
      location: { search: '' },
    };
    const tree = renderDeep(<SectionHowItWorksHero {...heroProps} />);
    delete window.google;
    expect(tree).toMatchSnapshot();
  });
});
