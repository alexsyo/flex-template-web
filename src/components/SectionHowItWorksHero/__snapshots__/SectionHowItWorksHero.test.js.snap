// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHowItWorksHero matches snapshot 1`] = `
<div
  className=""
>
  <div>
    <h1>
      <span>
        SectionHowItWorksHero.title
      </span>
    </h1>
    <h2>
      <span>
        SectionHowItWorksHero.subTitle
      </span>
    </h2>
    <Link
      className=""
      style={Object {}}
      title={null}
      to={
        Object {
          "pathname": "/s",
          "search": "address=Netherlands&bounds=53.665238%2C7.230902%2C50.750667%2C3.1862592",
        }
      }
    >
      <LinkAnchor
        className=""
        href="/s?address=Netherlands&bounds=53.665238%2C7.230902%2C50.750667%2C3.1862592"
        navigate={[Function]}
        style={Object {}}
        title={null}
      >
        <a
          className=""
          href="/s?address=Netherlands&bounds=53.665238%2C7.230902%2C50.750667%2C3.1862592"
          onClick={[Function]}
          style={Object {}}
          title={null}
        >
          <span>
            SectionHowItWorksHero.browseButton
          </span>
        </a>
      </LinkAnchor>
    </Link>
  </div>
</div>
`;
